### PR TITLE
Add level property to ancestors

### PR DESCRIPTION
**Pull request by akolson**
## Overview
This pr adds a level property to the ancestors property in the embed topics schema. `level` is required when a topic has ancestors.

### CODE REVIEW

File: le_utils/constants/embed_topics_request.py [hunk 1]
```diff
@@ -16,30 +16,50 @@
...
+            "description": "The description of the topic",
         },
         "language": {
             "type": "string",
             "description": "Language code from https://github.com/learningequality/le-utils/blob/main/le_utils/resources/languagelookup.json",
             "pattern": "^[a-z]{2,3}(?:-[a-zA-Z]+)?$",
         },
+        "level": {
+            "type": "integer",
+            "description": "The level of the ancestor, where the parent is 1 and the root is the highest level",
```
Comment (bjester) [inline le_utils/constants/embed_topics_request.py:35]: When you proposed this, I assumed you would make root the lowest `level`. Having it be the highest means the `level`s always depend on the folder depth at which you're looking up recommendations, meaning you have to recompute the levels on the ancestors in order to send it. 

Django MPTT, which we use on Studio for the hierarchical structure of content nodes, adds a `level` field to the models. In that situation, `level=0` is the root and so on increases as you go deeper in the hierarchy. This `level` being in the inverse of that is likely to cause confusion with those who are familiar with that.

All that said, the root node of a studio channel wouldn't be needed, since the root node of a channel is essentially the channel itself, and we don't use the channel title in the topic representation. So if you make `level` start at 0, saying 0 is the root node gives the impression it's the root channel node. So if that's the case, if a `level=0` was set, it could be ignored, otherwise it always uses level >=1

Overall, reversing the level would better align with our ecosystem's content channel architecture.

If you have strong feelings, an alternative would be to add the `parent` field to each. Then it's very clear which node is the root, because it's the node without a parent. But I don't think that's necessary and it makes sorting the ancestors a little more complicated than a simple `level` does.

**Review by bjester — CHANGES_REQUESTED** [review]
TL;DR - Having the highest `level` could lead to confusion, and wasn't something I thought you intended when you mentioned this on Slack

File: le_utils/constants/embed_topics_request.py (hunk 1 above)
Comment (akolson) [reply le_utils/constants/embed_topics_request.py:35]: Ahh okay, I see. I think this change shouldn't be much of a lift on either side(le-utils and curriculum-automation). I didn't think about the order too much, but I agree this is confusing for one coming in with knowledg on studio's hierarchical structure.

File: js/EmbedTopicsRequest.js [hunk 2]
```diff
@@ -10,34 +10,75 @@ export const SCHEMA = {
...
+      "description": "The description of the topic"
     },
     "language": {
       "type": "string",
       "description": "Language code from https://github.com/learningequality/le-utils/blob/main/le_utils/resources/languagelookup.json",
       "pattern": "^[a-z]{2,3}(?:-[a-zA-Z]+)?$"
     },
+    "level": {
+      "type": "integer",
+      "description": "The level of the ancestor, where the root is 0 and the parent is the highest level"
```
Comment (akolson) [inline js/EmbedTopicsRequest.js:32]: @bjester this should be better I guess?

Comment (akolson) [reply js/EmbedTopicsRequest.js:32]: The implementation of the breadcrumb logic is [here](https://github.com/learningequality/curriculum-automation/pull/3/files#diff-9cc36d0170f00e05923ae3a3ddafa96ff50894661b0c655582c51062617b6b00R14-R22) for context

Comment (bjester) [reply js/EmbedTopicsRequest.js:32]: Looks great!
